subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -125,6 +128,9 @@ public SnapshotHierarchy call(BuildOperationContext context) {
                                 statisticsSinceLastBuild.getRetainedDirectories(),
                                 statisticsSinceLastBuild.getRetainedMissingFiles()
                             );
+                            if (stateInvalidatedAtStartOfBuild) {
+                                LOGGER.warn("Parts of the virtual file system have been removed since they didn't support watching");
```

Comment:
```suggestion
                                LOGGER.warn("Parts of the virtual file system have been invalidated since they didn't support watching");
```

Reply:
I think "removed" is a bit weird here and could give the wrong ideas to less experienced users.